tests/h/traversal/annotation_test.py

```diff
@@ -9,340 +8,109 @@
 from h.traversal import AnnotationContext, AnnotationRoot
 
 
-class FakeGroup:
-    # NB: Tests that use this do not validate that the principals are correct
-    # for the indicated group. They validate that those principals are being
-    # transferred over to the annotation as expected
-    # As such, this has sort of a partial and wonky re-implementation of
-    # ``h.models.Group.__acl__``
-    # This is a symptom of the disease that is splitting ACL concerns between
-    # traversal/resources and model classes
-    # TODO: Refactor once we're able to move ACLs off of models
-    def __init__(self, principals):
-        acl = []
-        for p in principals:
-            acl.append((security.Allow, p, Permission.Group.READ))
-            if p == security.Everyone:
-                acl.append(
-                    (security.Allow, security.Authenticated, Permission.Group.FLAG)
-                )
-                acl.append(
-                    (security.Allow, security.Authenticated, Permission.Group.MODERATE)
-                )
-            else:
-                acl.append((security.Allow, p, Permission.Group.FLAG))
-                # Normally, the ``moderate`` permission would only be applied
-                # to the admin (creator) of a group, but this ``FakeGroup``
-                # is indeed fake. Tests in this module are merely around whether
-                # this permission is translated appropriately from a group
-                # to an annotation context (i.e. it should not be applied
-                # to private annotations)
-                acl.append((security.Allow, p, Permission.Group.MODERATE))
-        self.__acl__ = acl
```

Comment:
This is just very difficult to piece together when looking at the tests. It's not necessary for this object to mutate like this, we can just create the correct version in the tests that need it.

Reply:
I think a fake or host class can make sense, but as a rule of thumb if you'd normally write tests for it, it probably shouldn't be in the tests.